scripts/changelog/collect-package-changelogs.sh

```diff
@@ -0,0 +1,99 @@
+#!/bin/bash
+
+set -e
+[[ $DEBUG ]] && set -x
+
+if [ -z "${IMAGE_UUID}" ] ; then
+  echo "IMAGE_UUID env variable is mandatory"
+  exit 0
+fi
+
+BASE_PATH="/tmp"
+RPM_PACKAGE_LIST_PATH="${BASE_PATH}/rpm-packages.txt"
+RPM_PACKAGE_TAR_FILE_NAME="rpm-package-changelogs.tar.gz"
+RPM_PACKAGE_TAR_FILE_PATH="${BASE_PATH}/${RPM_PACKAGE_TAR_FILE_NAME}"
+CHANGELOG_DIRECTORY_NAME="${IMAGE_UUID}-changelogs"
+CHANGELOG_BASE_PATH="${BASE_PATH}/${CHANGELOG_DIRECTORY_NAME}"
+
+FILTERED_RPM_PACKAGES=("gpg-pubkey")
+
+function execute() {
+  echo "Cleaning up the temporary folders and files..."
+  clean_up
+  echo "Collecting installed rpm packages..."
+  collect_rpm_packages
+  echo "Storing changelogs to file by rpm package name..."
+  store_rpm_package_changelog_to_file
+  echo "Compressing the entire directory that contains generated files into ${RPM_PACKAGE_TAR_FILE_PATH} file..."
+  cd $BASE_PATH
+  tar -zcf $RPM_PACKAGE_TAR_FILE_NAME $CHANGELOG_DIRECTORY_NAME
+}
+
+function clean_up() {
+  if [ -f $RPM_PACKAGE_LIST_PATH ] ; then
+    rm $RPM_PACKAGE_LIST_PATH
```

Comment:
You can simplify file removes by using rm with -f flag so you won't need the check the file existence.
Or what I would more prefer: you can use a specific folder inside /tmp as BASE_PATH so you can simply delete the BASE_PATH folder itself.

Reply:
new specific folder introduced for storing all temporary files